### PR TITLE
doc: for llvmcall

### PR DESCRIPTION
**Pull request by ihnorton**
To get this started anyway.

### CODE REVIEW

**Comment by ihnorton** [comment]
#8919

File: doc/stdlib/ffi.rst [hunk 1]
```diff
@@ -248,3 +248,21 @@
...
+   block, arguments are available as consecutive unnamed SSA variables (%0, %1, etc.).
+
+   Note that the argument type tuple must be a literal tuple, and not a tuple-valued variable or expression.
+
+   Each ``ArgumentValue`` to ``llvmcall`` will be converted to the corresponding ``ArgumentType``,
+   by automatic insertion of calls to ``unsafe_convert(ArgumentType, cconvert(ArgumentType, ArgumentValue))``.
+   (see also the documentation for each of these functions for further details).
+   In most cases, this simply results in a call to ``convert(ArgumentType, ArgumentValue)``.
+
+   See ``test/llvmcall.jl`` for usage examples.
```
Comment (ivarne) [inline doc/stdlib/ffi.rst]: Does
`` `test/llvmcall.jl 
<https://github.com/JuliaLang/julia/blob/master/test/llvmcall.jl>`_`` work in this context?

**Comment by vchuravy** [comment]
So the changes for #11604 would be:

- llvmcall accepts as first argument a tuple of form (decl::String, IR::String), where decl can contain a single declaration or multiple. 

Also the second argument should be a of form `Tuple{Type1, ...}`

**Comment by jakebolewski** [comment]
Bump, it would seem useful to get this in.

**Comment by ihnorton** [comment]
Rebased.

**Comment by tkelman** [comment]
Needs to go in helpdb now I think?

**Comment by ihnorton** [comment]
I believe I have sacrificed the proper number of rare newts to satisfy the docsys god here. Also included other minor changes from running genstdlib.

There was an error about a missing shell script on travis, but I restarted and it seems to have proceeded this time.

File: doc/stdlib/collections.rst [hunk 2]
```diff
@@ -221,6 +221,12 @@ Iterable Collections
 
    Returns an array containing only the unique elements of the iterable ``itr``\ , in the order that the first of each set of equivalent elements originally appears. If ``dim`` is specified, returns unique regions of the array ``itr`` along ``dim``\ .
 
+   .. code-block:: julia
+
+       unique(f, itr)
+
+   Returns an array containing one value from ``itr`` for each unique value produced by ``f`` applied to elements of ``itr``\ .
+
```
Comment (yuyichao) [inline doc/stdlib/collections.rst]: Mind fixing this to become a proper function entry?

Comment (ihnorton) [inline doc/stdlib/collections.rst]: I tried, but I've hit my non-deterministic error rate-limit for the day. The system refuses to recognize the function signatures. (yes, I rebuilt Julia, matched the signatures in helpdb and collections.rst, rebuilt julia again, ...).

Also, every 3rd or 4th run of genstdlib I get:

```
ERROR: LoadError: UnicodeError: invalid character index
 [inlined code] from ./int.jl:187
```

Comment (ihnorton) [inline doc/stdlib/collections.rst]: (note that there are several different signatures, and the one in `set.jl` that takes an unspecified iterable is different from `unique(::AbstractArray, dim::Int)` definition in `multi.jl`. I tried a variety of different combinations, including splitting the doc strings up)

File: base/docs/basedocs.jl [hunk 3]
```diff
@@ -501,6 +501,22 @@ keywords[:ccall] = doc"""
   in a call to `convert(ArgumentType, ArgumentValue)`.
   """
 
+keywords[:llvmcall] = doc"""
+      llvmcall(IR::String, ReturnType, (ArgumentType1, ...), ArgumentValue1, ...)
+
+   Call LLVM IR string in the first argument. Similar to an LLVM function ``define``
```
Comment (tkelman) [inline base/docs/basedocs.jl]: default is markdown, so should be single backticks? though I think double backticks in markdown might be harmless

**Pull request merged**